[owenong1] iP

owenong1 opened the pull request:
# BotBot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

BotBot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines 
 - [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

owenong1 committed:
Renamed Chatbot, Implemented basic greet/exit

owenong1 committed:
Implemented echo feature, adjusted code structure

owenong1 committed:
Implemented task list, Minor edits to formatting

owenong1 committed:
Implemented marking; Added Task class and modified main logic

owenong1 committed:
Implemented Todo/Event/Deadline classes and 'list' command

owenong1 committed:
Auto testing and minor structure changes

owenong1 committed:
Implement some errors

owenong1 committed:
Implemented delete functionality

owenong1 committed:
Added save feature that saves to data/list.txt

owenong1 committed:
Merge branch 'branch-Level-7'

owenong1 committed:
Implemented Time, Added Exceptions to handle invalid time formats

owenong1 committed:
Restructured to use OOP

owenong1 committed:
Merge branch 'branch-A-MoreOOP'

owenong1 committed:
Re-factored classes into packages

owenong1 committed:
Merge commit '8a560b405243740a8b348e1efa85f561c0665449' into branch-Level-8

owenong1 committed:
Merge branch 'branch-Level-8'

owenong1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

owenong1 committed:
Added Junit tests

owenong1 committed:
Modify JAR settings

owenong1 committed:
Added javaDoc comments for relevant methods

owenong1 committed:
Implemented find command

ibnu2651 commented on src/test/java/botbot/task/TaskTest.java:
```diff
@@ -0,0 +1,20 @@
+
+public class TaskTest {
+    @Test
+    public void getRepToDoTest() {
```
I think for tests the naming convention is usually testToDoGetRep or something like that?

ibnu2651 commented on src/main/java/botbot/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+import java.time.format.DateTimeFormatter;
+public class Task {
+    String taskName;
+    Boolean done = false;
```
I'm not sure if this is within the scope of naming but I think for Java, Boolean is actually the wrapper class for the primitive value boolean which might be more suitable here? But if you meant to use the wrapper class then by all means.

ibnu2651 commented on src/main/java/botbot/exception/DescriptionException.java:
```diff
@@ -0,0 +1,7 @@
+package botbot.exception;
+
+public class DescriptionException extends BotBotException {
```
I think having a more descriptive name would help since DescriptionException does not really say what is wrong, maybe something like MissingDescriptionException would more clearly capture the nature of the error.

ibnu2651 commented on src/main/java/botbot/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+
+    /**
+     * Returns string format of a task
+     * @return String rep
```
While I think it's ok to use "rep" instead of "representation" for the method names, maybe you can use the full word for the Javadoc since it's supposed to describe the method in full.

ibnu2651 commented on src/main/java/botbot/BotBot.java:
```diff
@@ -0,0 +1,40 @@
+
+    /**
+     * Returns a new instance of chatbot
+     * @throws BotBotException
```
I think there should be an empty line between the method description and the params/throws/returns.

ibnu2651 reviewed (commented):
I think this is a pretty good effort that covers all the requirements. I was more concerned with the naming aspect of code quality and I could not find much fault so I raised some other concerns. Keep up the good work!

vannykin commented on src/main/java/botbot/Ui.java:
Good naming of all methods in all classes, very clean if, else-if, else segments, and clear logic.
Great readability! I don't see any coding standard violations :) yay

vannykin reviewed (commented):
Good job :) I should probably go look through mine now and improve my own readability

owenong1 committed:
Added checkstyle check compatability

owenong1 committed:
Merge branch 'branch-A-checkStyle'

owenong1 committed:
Basic GUI without functionality implemented

owenong1 committed:
Minor restructuring of files, GUI .xml without functionality implemented

owenong1 committed:
Implemented funcitonalities except for save/load

owenong1 committed:
Implemented Save/Load to the GUI implementation (GUI now fully functional)

owenong1 committed:
Merge branch 'branch-Level-10'

owenong1 committed:
Include some assert cases

There are some methods that can be tested with assert

These will be added

owenong1 committed:
Refined some code quality

Did trivial code refactoring as test

owenong1 committed:
Merge pull request #1 from owenong1/branch-A-Assertions

Include some assert cases

owenong1 committed:
no message

owenong1 committed:
Implemented priority

The code lacked a way to add priotity of tasks

This commit provides a basic implementation

owenong1 committed:
Merge branch 'branch-Extension' into branch-A-JavaDoc

owenong1 committed:
Merge branch 'branch-Level-9' into branch-A-CodeQuality

owenong1 committed:
Commit all final changes

Implementing user guide for all final changes

owenong1 committed:
Minor changes

File renaming

owenong1 committed:
Minor renaming

owenong1 committed:
no message

owenong1 committed:
Edited file path to be non-absolute

owenong1 committed:
Fixed classpath issue

Chenged the functions used in file creation to avoid errors
